### ISSUE TITLE
Monitor and fix OutOfRangeDateTimeExceptions

### ISSUE THREAD

**Issue by arturs105** [toggl/mobileapp#5237]
#5235 Introduced `OutOfRangeDateTimeException` to track the creation of date times which are out of range defined by Toggl business logic. That includes date times which cause unrepresentable date errors (Jan 1 0001)

We should check all the places, where out of range date times are created, especially date times that cause unrepresentable date errors.

**Comment by juanlaube** [comment]
I've been looking `OutOfRangeDateTimeException ` and `OutOfBoundsDateTimeCreatedException` in AppCenter and I cannot find anything, this doesn't seem to be happening anymore. I will wait for @arturs105 to return from vacations before closing the issue just in case he has another idea.

**Comment by arturs105** [comment]
Okay, seems like the problem was this: #5472. It's fixed and released in iOS 2.1, however only 1% of users have updated and the release is paused at this moment until we fix a sync issue, so we can't really know at the moment if it's actually fixed.

Now there is already some data in App Center from iOS 2.1 and the good news is that neither this nor other weird date related exceptions have occurred in 2.1. This might be because we have fixed the problem, but it could just as well be because not enough people have updated and we haven't received those exception yet.

Either way, moving this issue to a further release, when we will have more data to actually find out if we have fixed the problem.
